src/vm/gcenv.os.cpp

```diff
@@ -465,11 +480,9 @@ uint64_t GCToOSInterface::GetPhysicalMemoryLimit()
 {
     LIMITED_METHOD_CONTRACT;
 
-#ifndef FEATURE_PAL
     size_t restricted_limit = GetRestrictedPhysicalMemoryLimit();
-    if (restricted_limit != 0)
+    if (restricted_limit != 0 && restricted_limit != (size_t)MAX_PTR)
```

Comment:
Is the added check needed? It looks like the GetRestrictedPhysicalMemoryLimit should return zero if there is no limit to be on par with the Windows version.